boto/s3/key.py

```diff
@@ -230,6 +230,18 @@ def handle_encryption_headers(self, resp):
         else:
             self.encrypted = None
 
+    def handle_storage_class_header(self, resp):
+        provider = self.bucket.connection.provider
+        if provider.storage_class_header:
+            self._storage_class = resp.getheader(
+                provider.storage_class_header, None)
+            if (self._storage_class is None and
+                provider.get_provider_name() == 'aws'):
+                # S3 docs for HEAD object requests say S3 will return this
+                # header for all objects except Standard storage class objects.
```

Comment:
I don't understand why you want to populate the storage class in this case, if the S3 docs say they won't populate the header in this case?